dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/validation/hooks/RelationshipsValidationHook.java

```diff
@@ -93,10 +93,10 @@ public void validateRelationship( ValidationErrorReporter reporter, Relationship
     private void validateDuplication( ValidationErrorReporter reporter, Relationship relationship,
         TrackerBundle bundle )
     {
-        org.hisp.dhis.relationship.Relationship relationshipUsingKey = bundle.getPreheat()
-            .getRelationshipByKey( relationship );
+        org.hisp.dhis.relationship.Relationship duplicateRelationship = bundle.getPreheat()
+            .getDuplicatedRelationship( relationship );
 
-        if ( Objects.nonNull( relationshipUsingKey ) )
+        if ( Objects.nonNull( duplicateRelationship ) )
```

Comment:
so my point about not adding a getter would be to add something like `preheat.isDuplicate(Relationship)` instead

Reply:
Done